ambari-server/src/main/resources/stack-hooks/before-START/scripts/params.py

```diff
@@ -325,12 +325,13 @@
   namenode_rpc = default('/configurations/hdfs-site/dfs.namenode.rpc-address', default_fs)
 
 # if HDFS is not installed in the cluster, then don't try to access namenode_rpc
-if "core-site" in config['configurations'] and namenode_rpc:
-  port_str = namenode_rpc.split(':')[-1].strip()
-  try:
-    nn_rpc_client_port = int(port_str)
-  except ValueError:
-    nn_rpc_client_port = None
+if has_namenode and namenode_rpc:
+  if "core-site" in config['configurations'] and namenode_rpc:
```

Comment:
Can you please remove the duplicate `namenode_rpc` check, and collapse the nested `if`s into a single one?